packages/devtools_app/lib/src/standalone_ui/vs_code/debug_sessions.dart

```diff
@@ -184,11 +193,15 @@ class _DevToolsMenu extends StatelessWidget {
   }
 
   bool _shouldIncludeScreen(ScreenMetaData screen) {
-    // Don't show home screen or debugger in the menu.
+    // Some screens shouldn't show up in the menu.
     return screen != ScreenMetaData.home &&
         screen != ScreenMetaData.debugger &&
+        screen != ScreenMetaData.vmTools &&
         // Or the generic "simple" screen.
         screen != ScreenMetaData.simple &&
+        // DeepLink is currently behind a feature flag.
+        (screen != ScreenMetaData.deepLinks ||
```

Comment:
this should be ==

Reply:
I think it's correct - this is returning whether the screen should be included, and it can be included if it's not `deepLinks`, but if it is `deepLinks` we need the flag set.

Perhaps this multiline condition wasn't the simplest to read though, I've changed it to a switch expression which I think is much clearer.

Reply:
sorry I read this wrong. We might want to split this return statement up to make it more readable, and to make it less likely that logic bugs creep in. Something like this:

```dart
// Hide anything that requires some specific library that we can't check for.
if (screen.requiresLibrary != null) {
  return false;
}

return switch (screen) {
  ScreenMetaData.home ||
  ScreenMetaData.debugger ||
  ScreenMetaData.simple => false,
  ScreenMetaData.vmTools => preferences.vmDeverloperModeEnabled.value,
  ScreenMetaData.deepLinks => FeatureFlags.deepLinkValidation,
  _ => true,
};
```

Reply:
Returning `preferences.vmDeverloperModeEnabled.value` will only pick up the first value (e.g. won't listen for changes to the notifier), but my guess is that this setting is not something users turn on and off frequently. I expect advanced dart developers on the VM team will always just have this enabled, and the average user will not.

Reply:
Snap 😄

> `preferences.vmDeverloperModeEnabled.value`

I presume the preferences are loaded asynchronously, so if this menu was opened before that completed, this might not be right? Realistically that's unlikely though (you'd need to start a debug session before you could even get to the menu) so if you think it's reasonable to just read the current `value` I'll add this :-)